filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobFileSystemManager.scala

```diff
@@ -45,15 +48,27 @@ object BlobFileSystemManager {
   def hasTokenExpired(tokenExpiry: Instant, buffer: Duration): Boolean = Instant.now.plus(buffer).isAfter(tokenExpiry)
   def uri(endpoint: EndpointURL) = new URI("azb://?endpoint=" + endpoint)
 }
-case class BlobFileSystemManager(
-    container: BlobContainerName,
-    endpoint: EndpointURL,
-    expiryBufferMinutes: Long,
-    blobTokenGenerator: BlobSasTokenGenerator,
-    fileSystemAPI: FileSystemAPI = FileSystemAPI(),
-    private val initialExpiration: Option[Instant] = None) extends LazyLogging {
-  private var expiry: Option[Instant] = initialExpiration
+
+class BlobFileSystemManager(val container: BlobContainerName,
+                            val endpoint: EndpointURL,
+                            val expiryBufferMinutes: Long,
+                            val blobTokenGenerator: BlobSasTokenGenerator,
+                            val fileSystemAPI: FileSystemAPI = FileSystemAPI(),
+                            private val initialExpiration: Option[Instant] = None) extends LazyLogging {
+
+  def this(config: BlobFileSystemConfig) = {
+    this(
+      config.blobContainerName,
+      config.endpointURL,
+      config.expiryBufferMinutes,
+      BlobSasTokenGenerator.createBlobTokenGeneratorFromConfig(config)
```

Comment:
For my own education:

How are the fields `fileSystemAPI` and `initalExpiration` being initialized when this constructor is being used? Are they just using what's declared in the `class BlobFileSystemManager (...)` arguments?

Reply:
Yup, that's right! This line:
```val fileSystemAPI: FileSystemAPI = FileSystemAPI()```
means that the input is optional (not to be confused with being of type `Option`...), and if a caller doesn't specify it, the default `FilesystemAPI()` is used.